diff --git a/index.js b/index.js
index 004e0809024006335916314d605090629b73575f..210963e8c8befc89f8bfb6565bb77e9e16e6e0da 100644
--- a/index.js
+++ b/index.js
@@ -75,7 +75,8 @@ module.exports = function(options) {
   return function(Parser) {
     return plugin({
       allowNamespaces: options.allowNamespaces !== false,
-      allowNamespacedObjects: !!options.allowNamespacedObjects
+      allowNamespacedObjects: !!options.allowNamespacedObjects,
+			autoCloseVoidElements: !!options.autoCloseVoidElements
     }, Parser);
   };
 };
@@ -356,6 +357,13 @@ function plugin(options, Parser) {
         node.attributes.push(this.jsx_parseAttribute());
       node.selfClosing = this.eat(tt.slash);
       this.expect(tok.jsxTagEnd);
+			const VOID_ELEMENTS = [
+				'area', 'base', 'br', 'col', 'embed', 'hr', 'img', 'input', 'keygen',
+				'link', 'menuitem', 'meta', 'param', 'source', 'track', 'wbr'
+			];
+			if (options.autoCloseVoidElements && nodeName && VOID_ELEMENTS.includes(nodeName.name)) {
+				node.selfClosing = true;
+			}
       return this.finishNode(node, nodeName ? 'JSXOpeningElement' : 'JSXOpeningFragment');
     }
 
